### DISCUSSION TITLE
Runtime Train/Test dataset split questions

### DISCUSSION THREAD

**Discussion by YiKangOY** [pyg-team/pytorch_geometric#9745] [Q&A]
Hi! I have some graphs and I want to put some of them into training set and some into test set at runtime. 
Now I create two datasets, one for training and one for testing.
`train_dataset = MyOwnDataset('dataset_train')` 
`test_dataset = MyOwnDataset('dataset_test')`
But these requires building dataset everytime I change the train and test split.
Is there any way to put all graphs into one dataset and obtain training set and test set at runtime?

P.S. train_dataset= dataset[0:10] + dataset[20:30] is not a good choice because this will make train_dataset a _ConcatDataset_ object and can not directly access the properties of Pyg datasets like dataset.x

![image](https://github.com/user-attachments/assets/5226b500-9366-47d2-8d7c-6ccbb3d2c4b8)
![image](https://github.com/user-attachments/assets/a7f1577d-f423-4646-9c24-7bd1a5e5e0db)
